File: client/src/app/modules/tiers/services/sales.complain.service.ts
import { HttpClient, HttpParams } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { AppRestService } from 'src/app/services/appRest.service';
import { SalesComplain } from '../../miscellaneous/model/SalesComplain';

@Injectable({
  providedIn: 'root'
})
export class SalesComplainService extends AppRestService<SalesComplain>{

  constructor(http: HttpClient) {
    super(http, "tiers");
  }

  getComplainsByTiersId(tiersId: number) {
    return this.getById("sales-complains", tiersId);
  }

   addOrUpdateComplain(salesComplain: SalesComplain) {
    return this.addOrUpdate(new HttpParams(), "sales-complain", salesComplain, "Enregistré", "Erreur lors de l'enregistrement");
  }
}

File: src/main/java/com/jss/osiris/modules/osiris/miscellaneous/service/SalesComplainService.java
package com.jss.osiris.modules.osiris.miscellaneous.service;

import java.util.List;

import com.jss.osiris.modules.osiris.miscellaneous.model.SalesComplain;

public interface SalesComplainService {

        public List<SalesComplain> getComplains();

        public List<SalesComplain> getComplainsByTiersId(Integer idTiers);

        public SalesComplain addOrUpdateComplain(SalesComplain salesComplain);
}
